src/puppetlabs/puppetdb/cli/shovel.clj

```diff
@@ -0,0 +1,378 @@
+  and enqueues them into the normal queue"
+  [^ConnectionFactory conn-pool endpoint process-message]
+  (with-open [connection (create-connection conn-pool)
+              session (create-session connection)]
+    (let [request-browse (.createTopic session ScheduledMessage/AMQ_SCHEDULER_MANAGEMENT_DESTINATION)
+          browse-dest (.createTemporaryQueue session)]
+      (with-open [producer (.createProducer session request-browse)
+                  consumer (.createConsumer session browse-dest)]
+        (let [request (doto (.createMessage session)
+                        (.setStringProperty ScheduledMessage/AMQ_SCHEDULER_ACTION
+                                            ScheduledMessage/AMQ_SCHEDULER_ACTION_BROWSE)
+                        (.setJMSReplyTo browse-dest))
+              process-message-and-remove (fn [^Message msg]
+                                           (try
+                                             (process-message msg)
+                                             (let [msg-id (.getStringProperty msg
+                                                                              ScheduledMessage/AMQ_SCHEDULED_ID)
+                                                   remove (doto (.createMessage session)
+                                                            (.setStringProperty ScheduledMessage/AMQ_SCHEDULER_ACTION
+                                                                                ScheduledMessage/AMQ_SCHEDULER_ACTION_REMOVE)
+                                                            (.setStringProperty ScheduledMessage/AMQ_SCHEDULED_ID msg-id))]
+                                               (.send producer remove))
+                                             (catch Exception e nil)))]
+          (.send producer request)
+          (Thread/sleep 2000)
+          (log/info (i18n/trs "Processing messages from the Scheduler"))
+          (consume-everything consumer process-message-and-remove))))))
+
+(def default-mq-endpoint "puppetlabs.puppetdb.commands")
+(def retired-mq-endpoint "com.puppetlabs.puppetdb.commands")
+
+(defn- set-usage!
+  "Internal helper function for setting `SystemUsage` values on a `BrokerService`
+  instance.
+
+  `broker`    - the `BrokerService` instance
+  `megabytes` - the value to set as the limit for the desired `SystemUsage` setting
+  `usage-fn`  - a function that accepts a `SystemUsage` instance and returns
+  the child object whose limit we are configuring.
+  `desc`      - description of the setting we're configuring, to be used in a log message
+  "
+  [^BrokerService broker
+   megabytes
+   usage-fn
+   ^String desc]
+  {:pre [((some-fn nil? integer?) megabytes)
+         (fn? usage-fn)]}
+  (when-let [limit (some-> megabytes (* 1024 1024))]
+    (log/info (i18n/trs "Setting ActiveMQ {0} limit to {1} MB" desc megabytes))
+    (-> (.getSystemUsage broker)
+        usage-fn
+        (.setLimit limit)))
+  broker)
+
+(defn ^:dynamic enable-jmx
```

Comment:
As discussed, might want to check to make sure we're not actually firing up amqp jmx if we didn't mean to,